dspace-api/src/main/java/org/dspace/content/dao/impl/CollectionDAOImpl.java

```diff
@@ -159,7 +159,7 @@ public List<Collection> findAuthorizedByGroup(Context context, EPerson ePerson,
 
     @Override
     public List<Collection> findCollectionsWithSubscribers(Context context) throws SQLException {
-        return list(createQuery(context, "SELECT DISTINCT col FROM Subscription s join  s.collection col"));
+        return list(createQuery(context, "SELECT DISTINCT col FROM Subscription s join  s.dSpaceObject col"));
```

Comment:
can this be a performance bottleneck?

Reply:
Performance bottleneck? Why performance? This change of code fixed UserCheck without this fix that check isn't working.

Reply:
This method is used only by UserCheck, so it won't affect any other functionality.